### ISSUE TITLE
False positive for implicit_getter violation with async getters

### ISSUE THREAD

**Issue by StarLard** [realm/SwiftLint#3689]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Describe the bug

Using `get async` incorrectly triggers an implicit_getter violation: "Implicit Getter Violation: Computed read-only properties should avoid using the get keyword. (implicit_getter)"

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
warning: Implicit Getter Violation: Computed read-only properties should avoid using the get keyword. (implicit_getter)
```

### Environment

* SwiftLint version (run `swiftlint version` to be sure)? 0.43.1
* Installation method used (Homebrew, CocoaPods, building from source, etc)? CocoaPods
* Paste your configuration file: NA

* Are you using [nested configurations](https://github.com/realm/SwiftLint#nested-configurations)?
  If so, paste their relative paths and respective contents. No
* Which Xcode version are you using (check `xcodebuild -version`)? Xcode 13 beta 4
* Do you have a sample that shows the issue? Run `echo "[string here]" | swiftlint lint --no-cache --use-stdin --enable-all-rules`
  to quickly test if your example is really demonstrating the issue. If your example is more
  complex, you can use `swiftlint lint --path [file here] --no-cache --enable-all-rules`.

```swift
var someValue: Int {
    get async { // This triggers a violation
        ....
    }
}
```

**Comment by rudedogdhc** [comment]
This is also triggered with

```
var someValue: SomeType {
  get throws {
    ...
  }
}
```
